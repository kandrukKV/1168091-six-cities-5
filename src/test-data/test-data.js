export const noon = () => {};

export const additions = [
  `addition1`,
  `addition2`,
  `addition3`,
  `addition4`,
  `addition5`,
];

export const cities = [
  `Amsterdam`,
  `Paris`,
];

export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: false,
    rating: 1,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.902976,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 2,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.388540000000006,
      longitude: 4.899976,
      zoom: 16
    },
    id: 2
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: true,
    rating: 3,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.364540000000005,
      longitude: 4.9019759999999994,
      zoom: 16
    },
    id: 3
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: false,
    rating: 4,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.886976,
      zoom: 16
    },
    id: 4
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 5,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16
    },
    id: 5
  }
];

export const mosckReviews = [
  {
    id: 1,
    user: {
      id: 18,
      name: `Sophie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`
    },
    rating: 3,
    comment: `We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)`,
    date: `2020-11-02T13:38:44.752Z`
  },
  {
    id: 2,
    user: {
      id: 16,
      name: `Mollie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`
    },
    rating: 2,
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-11-02T13:38:44.752Z`
  }
];
