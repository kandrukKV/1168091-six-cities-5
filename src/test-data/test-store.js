import configureStore from "redux-mock-store";
import {mockOffers, mockReviews, cities} from "../test-data/test-data";

const createStore = configureStore();

const mockStore = createStore({
  API_DATA: {
    offers: mockOffers,
    favoriteOffers: mockOffers,
    offerDetails: {
      card: mockOffers[0],
      reviews: mockReviews,
      nearPlaces: mockOffers
    }
  },
  APP_STATE: {
    currentCity: `Amsterdam`,
    cities,
    currentSortType: `Popular`,
    activeCard: null,
    reviewFormState: `EDITING`
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    authInfo: null
  }
});

export {mockStore};
