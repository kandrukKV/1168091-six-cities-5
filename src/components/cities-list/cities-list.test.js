import React from "react";
import renderer from "react-test-renderer";
import CitiesList from "./cities-list";
import {BrowserRouter as Router} from 'react-router-dom';
import {cities, noon} from "../../test-data/test-data";

const currentCity = cities[0];

it(`CitiesList is rendered correctly`, () => {
  const tree = renderer.create(
      <Router>
        <CitiesList
          cities={cities}
          currentCity={currentCity}
          changeCity={noon}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
