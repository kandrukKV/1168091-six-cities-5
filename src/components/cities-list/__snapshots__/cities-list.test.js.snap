// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList is rendered correctly 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>,
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            aria-current="page"
            className="locations__item-link tabs__item tabs__item--active active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            aria-current="page"
            className="locations__item-link tabs__item  active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>,
]
`;
